### PR TITLE
feat: allow admin key and admin api cert to be stored in vault

### PR DESCRIPTION
**Pull request by rodman10**
### Description

Fixes https://github.com/apache/apisix/issues/9915

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by Sn0rt** [comment]
the TEST CI look good

**Comment by Sn0rt** [comment]
If you think you can enter the review, or you want to run a CI test first. You can remove the Draft label.

Generally, I will conduct code review after CI is completed.

**Comment by Sn0rt** [comment]
@rodman10 are you still process this PR ?

**Comment by rodman10** [comment]
I have removed the draft state and ready for review.@Sn0rt

**Comment by Sn0rt** [comment]
> I have removed the draft state and ready for review.@Sn0rt

PTAL the CI ?

**Comment by rodman10** [comment]
> > I have removed the draft state and ready for review.@Sn0rt
> 
> PTAL the CI ?

Ok, I will check it.

**Comment by rodman10** [comment]
> > I have removed the draft state and ready for review.@Sn0rt
> 
> PTAL the CI ?

I have looked into the `centos7` case, 

企业微信截图_16920688186997

It seems failed in case `t/deployment/conf_server.t`, but I can't reproduce the problem with cmd `prove -I../test-nginx/lib -I./ -r -s t/deployment/conf_server.t`, whether I have mistaken the test method @Sn0rt ? 

企业微信截图_16920685157714

**Comment by Sn0rt** [comment]
@rodman10 Don't worry, sometimes ci is unstable. Re-run CI to try.

**Comment by Sn0rt** [comment]
Can I cooperate? Give me permission

**Comment by rodman10** [comment]
> Can I cooperate? Give me permission

Ok.

File: t/cli/test_admin_mtls.sh [hunk 1]
```diff
@@ -53,3 +53,45 @@ if [ ! "$code" -eq 400 ]; then
...
+
+# correct certs
+code=$(curl -i -o /dev/null -s -w %{http_code}  --cacert ./t/certs/mtls_ca.crt --key ./t/certs/mtls_client.key --cert ./t/certs/mtls_client.crt -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' https://admin.apisix.dev:9180/apisix/admin/routes)
+if [ ! "$code" -eq 200 ]; then
+    echo "failed: failed to enabled mTLS for admin with vault"
+    exit 1
+fi
+
+# skip
+code=$(curl -i -o /dev/null -s -w %{http_code} -k -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' https://admin.apisix.dev:9180/apisix/admin/routes)
```
Comment (Sn0rt) [inline t/cli/test_admin_mtls.sh:91]: this test case can't pass

**Review by Sn0rt — CHANGES_REQUESTED** [review]

File: t/cli/test_admin_mtls.sh [hunk 2]
```diff
@@ -53,3 +53,45 @@ if [ ! "$code" -eq 400 ]; then
...
+            admin_ssl_cert_key: "$secret://apisix_config/admin_ssl_cert_key"
+            admin_ssl_ca_cert: "$secret://apisix_config/admin_ssl_ca_cert"
+    secret_vault:
+        enable: true
+        uri: "http://127.0.0.1:8200"
+        prefix: "kv/apisix"
+        token: "${{VAULT_TOKEN}}"
+' > conf/config.yaml
+
+VAULT_TOKEN='root' VAULT_ADDR='http://0.0.0.0:8200' vault kv put kv/apisix/apisix_config admin_ssl_cert=@./t/certs/mtls_server.crt admin_ssl_cert_key=@./t/certs/mtls_server.key admin_ssl_ca_cert=@./t/certs/mtls_ca.crt
```
Comment (Sn0rt) [inline t/cli/test_admin_mtls.sh:76]:
```diff
diff --git a/.github/workflows/cli.yml b/.github/workflows/cli.yml
index 7aa65540..4cde0dfe 100644
--- a/.github/workflows/cli.yml
+++ b/.github/workflows/cli.yml
@@ -53,6 +53,7 @@ jobs:
       - name: Linux launch common services
         run: |
           project_compose_ci=ci/pod/docker-compose.common.yml make ci-env-up
+          sudo ./ci/init-common-test-service.sh
```

**Comment by kingluo** [comment]
The code is fine with me.

But I think it should be noted in the PR description which configuration fields support encryption:
* [x] Admin Key
* [x] Admin API certificate
* [ ] etcd password
* [ ] etcd certificate
* [ ] information that may exist in the plugin_attr

Also, whether in this PR or not, the relevant documentation should also describe this change.

**Comment by Sn0rt** [comment]
> The code is fine with me.
> 
> But I think it should be noted in the PR description which configuration fields support encryption:
> 
> * [x]  Admin Key
> * [x]  Admin API certificate
> * [ ]  etcd password
> * [ ]  etcd certificate
> * [ ]  information that may exist in the plugin_attr
> 
> Also, whether in this PR or not, the relevant documentation should also describe this change.

yep. Because our underlying library can only use the path to pass in the certificate, so avoid the PR being too large. We will support this feature in other PRs.

**Review by Sn0rt — APPROVED** [review]

**Comment by Sn0rt** [comment]
I plan to close this PR and continue to improve this feature based on your commit and new proposal.
The new PR will contain complete information about your commit.
Do you have any suggestions? 

@rodman10

**Comment by rodman10** [comment]
> I plan to close this PR and continue to improve this feature based on your commit and new proposal. The new PR will contain complete information about your commit. Do you have any suggestions?
> 
> @rodman10

Ok.

**Pull request closed without merging**